One new C function, several documentation changes

plredmond opened the pull request:
Hi! This is a random grab-bag of documentation improvements. In addition to that, I've exposed exactly one additional C API function.

plredmond commented on default.nix:
```diff
@@ -0,0 +1,29 @@
+  haskellPackages = haskellCompilerPackages.override
+    (old: {
+      overrides = self: super: with nixpkgs.haskell.lib; rec {
+        gomp = self.hello;
```
This is definitely not correct for OSX or MacOS users. It might be better to leave out `default.nix` and `makefile` from the PR.

plredmond replied:
They are deleted.

IagoAbal reviewed (commented):
Yes please leave `.gitignore`, `default.nix`, and `makefile` out of the PR. But I am happy to take the other changes. Thanks!

plredmond commented:
I've reverted the commit which added those files, and also cherry-picked another commit which adds a few more functions. I don't know if I added them in the "right" place vertically in the file, but otherwise followed the conventions.

IagoAbal reviewed (approved):
Thanks!
